locallib.php

```diff
@@ -112,6 +112,16 @@ function report_customsql_generate_csv($report, $timenow) {
         fclose($handle);
     }
 
+    // Copy the file to the custom directory.
+    if (!empty($report->customdir)) {
+        list($csvfilename, $csvtimestamp) = report_customsql_csv_filename($report, $timenow);
+        $filename = $report->id . '-' . basename($csvfilename);
+        // Make sure we always have a working path.
+        $filepath = rtrim($report->customdir, '/');
+        $filepath = $filepath . '/' . $filename;
+        copy($csvfilename, $filepath);
+    }
+
```

Comment:
I think you should move this chunk to be next to the email bit, after the update record. That way, an error in copying the file does not stop the DB being updated.

Also, following the style around here, the chunk of cdoe inside the if statement should probably be extracted into a report_customsql_copy_csv_to_customdir. (You may be able to think of a better name.)